### PR TITLE
[7884] Solving RTPS->DDS migration error on Log namespace

### PR DESCRIPTION
**Pull request by MiguelBarro**
This error is only perceptible building on debug mode because involves the macro definitions. The following works were done:
+ Migrate all includes from rtps to dds namespace.
+ Update the mocks to match the new dds tree structure.
+ Update the macros to use namespace resolution. When full dds migration is accomplished this change can be actually undone.

### CODE REVIEW

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1523/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1295/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1521/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1525/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1297/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1523/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1527/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1299/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1525/)

File: test/unittest/xmlparser/XMLProfileParserTests.cpp [hunk 1]
```diff
@@ -29,6 +29,10 @@ using namespace eprosima::fastrtps;
 using namespace eprosima::fastrtps::rtps;
 using namespace ::testing;
 
+using eprosima::fastdds::dds::Log;
+using eprosima::fastdds::dds::LogMock;
+using eprosima::fastdds::dds::LogConsumer;
+
 // Initialize Log mock
 LogMock* log_mock;
 std::function<void(std::unique_ptr<LogConsumer>&&)> Log::RegisterConsumerFunc =
```
Comment (MiguelCompany) [inline test/unittest/xmlparser/XMLProfileParserTests.cpp]: The build errors on VS2015 can be solved by creating the lambdas before this line (with auto), and assigning them later

Comment (MiguelBarro) [reply test/unittest/xmlparser/XMLProfileParserTests.cpp]: I've actually decided to make the lambdas disappeared by defining suitable functions on the source files.

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1542/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1314/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1540/)

**Pull request merged**